packages/shared/src/components/buttons/Button.tsx

```diff
@@ -50,7 +51,7 @@ export interface BaseButtonProps {
 const useGetIconWithSize = (size: ButtonSize, iconOnly: boolean) => {
   return (icon: React.ReactElement<IconProps>) =>
     React.cloneElement(icon, {
-      size: IconSize[size],
+      size: icon.props?.size ?? buttonSizeToIconSize[size],
```

Comment:
Had to add this as we have 1 icon (tags) being smaller then guidelines, asked design if this is correct.

Reply:
Oh, nice catch. Though it seems to be inaccurate in that case. Hope they will rectify it so we won't have to check this.

Reply:
Apparently the "Tag" is not a button from design perspective, checking with Tomer now how they want to proceed.

Reply:
Design needs to look into this and will later on let us know the right decision.
For now let's merge this and retro fix this later.